### PR TITLE
feat: [CI-5024]: Enable Hosted builds feature for `HOSTED_BUILDS` feature flag

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

Enable Hosted builds feature for `HOSTED_BUILDS` feature flag

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/hooks/useHostedBuild.ts [hunk 1]
```diff
@@ -8,18 +8,21 @@
...
 
 export function useHostedBuilds(): UseHostedBuildsReturn {
+  const { HOSTED_BUILDS } = useFeatureFlags()
   const { licenseInformation } = useLicenseStore()
   const { status, edition, licenseType } = licenseInformation['CI'] || {}
   // Hosted Builds to be only enabled for free users or paid enterprise/team users
   return {
-    enabledHostedBuildsForFreeUsers: edition === Editions.FREE && status === LICENSE_STATE_VALUES.ACTIVE,
+    enabledHostedBuildsForFreeUsers:
+      (edition === Editions.FREE && status === LICENSE_STATE_VALUES.ACTIVE) || HOSTED_BUILDS || false,
```
Comment (d1wilko) [inline src/modules/10-common/hooks/useHostedBuild.ts:25]: i think if you make this `enabledHostedBuildsForFreeUsers:
      HOSTED_BUILDS || (edition === Editions.FREE && status === LICENSE_STATE_VALUES.ACTIVE),`
you can get rid of the false at the end

**Review by d1wilko — CHANGES_REQUESTED** [review]
1 small change :)

File: src/modules/10-common/hooks/useHostedBuild.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/10-common/hooks/useHostedBuild.ts:25]: Good catch! Done.

**Review by d1wilko — APPROVED** [review]
LGTM

**Pull request merged**